### PR TITLE
Issue 96 - login failure now displays error to user

### PR DESCRIPTION
**Pull request by mfelix5**
This seems to fix the issue without breaking other login or login failure functionality. 

Review and suggestions are appreciated because I don't fully understand why componentWillReceiveProps was not being called or the solution. When I was looking into it, I saw that cWRP is deprecated and I replaced it with getDerivedStateFromProps. That said, it looks like Facebook recommends against this pattern (whether cWRP or gDSFP), something that is used throughout the codebase.

https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops
https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

### CODE REVIEW

File: screens/login-screen/login.js [hunk 1]
```diff
@@ -77,7 +76,12 @@ class Login extends Component<Props> {
         title: 'Log In'
     };
 
-    componentWillReceiveProps(nextProps) {
+    constructor(props) {
+        super(props);
+        this.state = {};
+    }
+
+    static getDerivedStateFromProps(nextProps) {
```
Comment (smara-codes) [inline screens/login-screen/login.js]: I'm pretty sure you could move this into the render function. The logic is: if there's an error then alert else render the login screen - right?

My understanding is that we'd use getDerivedStateFromProps if we needed to update the component internal state based on updates in props which we are not.

https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state

Comment (mfelix5) [reply screens/login-screen/login.js]: good point. I'll do that.

**Comment by mfelix5** [comment]
@smaraf , I moved to render. Let me know what you think. Thanks!

**Review by smara-codes — APPROVED** [review]
Looks good @mfelix5!

**Pull request merged**